gridcapa-cse-valid-app/src/main/java/com/farao_community/farao/cse_valid/app/ZonalScalableProvider.java

```diff
@@ -0,0 +1,111 @@
+
+    public ZonalScalableProvider(FileImporter fileImporter, EicCodesConfiguration eicCodesConfiguration) {
+        this.fileImporter = fileImporter;
+        this.eicCodesConfiguration = eicCodesConfiguration;
+    }
+
+    public ZonalData<Scalable> get(String glskUrl, Network network, ProcessType processType) {
+        ZonalData<Scalable> zonalScalable = fileImporter.importGlsk(glskUrl).getZonalScalable(network);
+        eicCodesConfiguration.getCseCodes().forEach(eic -> checkCseCountryInGlsk(zonalScalable, eic));
+        stackScalableOnLoads(network, zonalScalable, processType);
+        return zonalScalable;
+    }
+
+    private static void checkCseCountryInGlsk(ZonalData<Scalable> zonalScalable, String eic) {
+        if (!zonalScalable.getDataPerZone().containsKey(eic)) {
+            throw new CseValidInvalidDataException(String.format("Area '%s' was not found in the glsk file.", eic));
+        }
+    }
+
+    private void stackScalableOnLoads(Network network, ZonalData<Scalable> zonalScalable, ProcessType processType) {
+        zonalScalable.getDataPerZone().forEach((eiCode, scalable) -> {
+            if (processType == ProcessType.IDCC && eiCode.equals(eicCodesConfiguration.getItaly())) {
+                return;
+            }
+            double sum = getZoneSumOfActiveLoads(network, eiCode);
+            // No need to go further if a country has no active load
+            if (sum == 0.0) {
+                return;
+            }
+            Scalable stackedScalable = getStackedScalable(eiCode, scalable, network, sum);
+            zonalScalable.getDataPerZone().put(eiCode, stackedScalable);
+        });
+    }
+
+    private double getZoneSumOfActiveLoads(Network network, String cseCountry) {
+        return network.getLoadStream()
+            .filter(load -> isLoadCorrespondingToTheCountry(load, cseCountry))
+            .map(Load::getP0)
+            .reduce(0., Double::sum);
+    }
+
+    private Scalable getStackedScalable(String cseCountry, Scalable scalable, Network network, double sum) {
+        List<Float> percentageList = new ArrayList<>();
+        List<Scalable> scalableList = new ArrayList<>();
+
+        network.getLoadStream()
+            .filter(load -> isLoadCorrespondingToTheCountry(load, cseCountry))
+            .forEach(load -> {
+                percentageList.add((float) (load.getP0() / sum) * 100);
+                scalableList.add(Scalable.onLoad(load.getId()));
+            });
+
+        return Scalable.stack(scalable, Scalable.proportional(percentageList, scalableList, true));
+    }
+
+    private boolean isLoadCorrespondingToTheCountry(Load load, String cseCountry) {
+        return load.getTerminal().getVoltageLevel().getSubstation()
+            .flatMap(Substation::getCountry)
+            .map(this::getEicFromCountry)
+            .map(cseCountry::equals)
+            .orElse(false);
+    }
+
+    private String getEicFromCountry(Country country) {
```

Comment:
put this method in the EicCodesMapper or use mapToEicCodes(String countryStr)

Reply:
Will move it to the mapper class and factorize with existing `mapToEicCodes(String countryStr)` method.